### PR TITLE
IDOR: Support column-to-column comparisons

### PR DESCRIPTION
**Pull request by timokoessler**
## Summary by Aikido
|  Security Issues: 0 | 🔍 Quality Issues: 3 |  Resolved Issues: 0 |
| :--- | :--- | :--- |

**⚡ Enhancements**
* Collected and propagated column-to-column comparisons across JOIN chains.
* Resolved column-to-column filters for UPDATE and DELETE queries.
* Added tests covering column-to-column resolution, joins, and edge cases.

**🔧 Refactors**
* Refactored filter logic: replaced tuples with structs and helpers.

<sup>[More info](https://app.aikido.dev/featurebranch/scan/133550029?groupId=6)</sup>

### CODE REVIEW

File: src/idor/idor_analyze_sql.rs [hunk 1]
```diff
@@ -634,7 +636,9 @@ fn collect_col_col_pairs_from_joins(tables: &[TableWithJoins]) -> Vec<ColColPair
     let mut pairs = Vec::new();
     for twj in tables {
         for join in &twj.joins {
-            if let JoinOperator::Inner(JoinConstraint::On(on_expr)) = &join.join_operator {
+            if let JoinOperator::Join(JoinConstraint::On(on_expr))
```
Comment (timokoessler) [inline src/idor/idor_analyze_sql.rs:639]: This still does not match a LEFT or RIGHT JOIN, we have tests on it

**Review by tomaisthorpe — APPROVED** [review]

**Review by hansott — APPROVED** [review]
